flink-connectors/flink-connector-aws-kinesis-data-streams/src/main/java/org/apache/flink/connector/kinesis/sink/KinesisDataStreamsSinkBuilder.java

```diff
@@ -105,7 +118,10 @@
     @Override
     public KinesisDataStreamsSink<InputT> build() {
         return new KinesisDataStreamsSink<>(
-                getElementConverter(),
+                new KinesisDataStreamsSinkElementConverter.Builder<InputT>()
+                        .setSerializationSchema(serializationSchema)
+                        .setPartitionKeyGenerator(partitionKeyGenerator)
+                        .build(),
```

Comment:
Do we need to validate `serializationSchema` and `partitionKeyGenerator` are non null here?

Reply:
We're doing those validation checks in `KinesisDataStreamsSinkElementConverter`... would it be sufficient do you think?